src/pages/ListPets/components/PetGallery/index.tsx

```
@@ -51,72 +54,260 @@ export default function PetGallery() {
-    return true;
-  });
+        console.log("Full API response:", response);
 
-  const totalPages = Math.ceil(filteredPets.length / 6);
+        // Extract pets from the response format
+        let pets: Pet[] = [];
+        let totalItems = 0;
+
+        // Handle the specific response format from your API
+        if (response.data && response.data.success) {
+          // The API returns { statusCode, success, message, data: { items: Pet[], totalCount, pageNumber, pageSize }, detailErrors }
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)